src/server/journal/streamer.cc

```diff
@@ -79,7 +81,9 @@ void JournalStreamer::Cancel() {
   VLOG(1) << "JournalStreamer::Cancel";
   waker_.notifyAll();
   journal_->UnregisterOnChange(journal_cb_id_);
-  WaitForInflightToComplete();
+  if (!cntx_->IsCancelled()) {
+    WaitForInflightToComplete();
```

Comment:
I was about to write that maybe we should move `cntx_->IsCancelled()` within `WaitForInFlightToComplete` but then I realized it's only called in this place so not really needed I guess